saved-sites/saved-sites-impl/src/main/java/com/duckduckgo/savedsites/impl/sync/SavedSitesSyncPersister.kt

```diff
@@ -142,6 +153,31 @@ class SavedSitesSyncPersister @Inject constructor(
         savedSitesRepository.pruneDeleted()
     }
 
+    private fun updateSavedSitesMetadataWhenNoRemoteChanges() {
+        // for all items in the metadata table
+        // copy request columns to children
+        // delete request column values
+        Timber.d("Sync-Bookmarks-Metadata: set metadata for all local folders")
+        savedSitesSyncRepository.confirmAllFolderChildrenMetadata()
+    }
+
+    private fun updateSavedSitesMetadataWhenRemoteChanges(entities: List<SyncBookmarkEntry>) {
+        // for all items in the payload
+        // add children to children column
+        // delete request column values
+
+        // for all items in the metadata table and not in the payload
+        // copy request columns to children
+        // delete request column values
+        if (entities.isEmpty()) {
```

Comment:
I'm confused because we have "metadata" logic in the persister and in the Persister Algorithm, and it seems duplicated.

And both seems to store data from the payload, whereas only this one will consider persisting the data not present in the payload. So there's a difference, but 2 places persist data from the payload.

Which class do you think should own metadata logic?

Reply:
fixed